pkg/aws/eni/limits/limits.go

```diff
@@ -883,3 +926,65 @@ func parseLimitString(limitString string) (limit ipamTypes.Limits, err error) {
 	}
 	return ipamTypes.Limits{Adapters: intSlice[0], IPv4: intSlice[1], IPv6: intSlice[2]}, nil
 }
+
+func newLimitsUpdateTrigger(api ec2API, minInterval, timeout time.Duration, retryCount int) (*trigger.Trigger, error) {
+	return trigger.NewTrigger(trigger.Parameters{
+		Name:        "ec2-instance-limits-update",
+		MinInterval: minInterval,
+		TriggerFunc: func(reasons []string) {
+			if err := updateFromEC2APIWithRetry(context.TODO(), api, timeout, retryCount); err != nil {
```

Comment:
nit: Consider using a `context.WithCancel` here where the the `cancel` func is then assigned to `ShutdownFunc` in the trigger params.

We are currently never shutting down the trigger, but if we do in the future it will automatically kill any running request. Being able to shut down the trigger might also be useful in unit tests.